lib/marc_match_key/publisher_key.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the publisher portion of a GoldRush key
+  class PublisherKey
+    include MarcMatchFunctions
+    attr_reader :record
+
+    def initialize(record)
+      @record = record
+    end
+
+    def key
+      publisher_from_pub_field(publisher_26x_field)
+    end
+
+    private
+
+    def publisher_26x_field
+      publisher_from_f264 || record.fields('260').find { |f| f['b'] }
+    end
+
+    def publisher_from_f264
+      f264 = record.fields('264').select { |f| f['b'] }
+      preferred_order = %w[1 4 2 3 0]
+      preferred_order.each do |indicator|
+        field = f264.find { |f| f.indicator2 == indicator }
+        return field if field
+      end
+      nil
+    end
+
+    def publisher_from_pub_field(pub_field)
+      return pad_with_underscores('', 5) if pub_field.nil?
+
+      subfb = pub_field['b'].dup.to_s
```

Comment:
Since the `pub_field` parameter will only ever be `publisher_26x_field`, could we remove the parameter?  And we could memoize the `publisher_26x_field` method so that it wouldn't result in another trip through the MARC record.

Also, we don't need the `#dup`

```suggestion
    def publisher_from_pub_field
      return pad_with_underscores('', 5) if publisher_26x_field.nil?

      subfb = publisher_26x_field['b'].to_s
```